pkg/container/types/tuple.go

```diff
@@ -0,0 +1,663 @@
+		case uint32:
+			res += fmt.Sprintf("(uint32: %v)", t)
+		case uint64:
+			res += fmt.Sprintf("(uint64: %v)", t)
+		case Date:
+			res += fmt.Sprintf("(date: %v)", t.String())
+		case Datetime:
+			res += fmt.Sprintf("(datetime: %v)", t.String())
+		case Timestamp:
+			res += fmt.Sprintf("(timestamp: %v)", t.String())
+		case Decimal64:
+			res += fmt.Sprintf("(decimal64: %v)", t.String())
+		case Decimal128:
+			res += fmt.Sprintf("(decimal128: %v)", t.String())
+		case []byte:
+			res += fmt.Sprintf("([]byte: %v)", t)
+		case float32:
+			res += fmt.Sprintf("(float32: %v)", t)
+		case float64:
+			res += fmt.Sprintf("(float64: %v)", t)
+		default:
+			res += fmt.Sprintf("(unorganizedType: %v)", t)
+		}
+		if i != len(tuple)-1 {
+			res += ","
+		}
+	}
+	res += ")"
+	return res
+}
+
+const nilCode = 0x00
+const bytesCode = 0x01
+const intZeroCode = 0x14
+const float32Code = 0x20
+const float64Code = 0x21
+const falseCode = 0x26
+const trueCode = 0x27
+const int8Code = 0x28
+const int16Code = 0x29
+const int32Code = 0x3a
+const int64Code = 0x3b
+const uint8Code = 0x3c
+const uint16Code = 0x3d
+const uint32Code = 0x3e
+const uint64Code = 0x40
+const dateCode = 0x41
+const datetimeCode = 0x42
+const timestampCode = 0x43
+const decimal64Code = 0x44
+const decimal128Code = 0x45
+const stringTypeCode = 0x46
+
+var sizeLimits = []uint64{
+	1<<(0*8) - 1,
+	1<<(1*8) - 1,
+	1<<(2*8) - 1,
+	1<<(3*8) - 1,
+	1<<(4*8) - 1,
+	1<<(5*8) - 1,
+	1<<(6*8) - 1,
+	1<<(7*8) - 1,
+	1<<(8*8) - 1,
+}
+
+func bisectLeft(u uint64) int {
+	var n int
+	for sizeLimits[n] < u {
+		n++
+	}
+	return n
+}
+
+func adjustFloatBytes(b []byte, encode bool) {
+	if (encode && b[0]&0x80 != 0x00) || (!encode && b[0]&0x80 == 0x00) {
+		// Negative numbers: flip all of the bytes.
+		for i := 0; i < len(b); i++ {
+			b[i] = b[i] ^ 0xff
+		}
+	} else {
+		// Positive number: flip just the sign bit.
+		b[0] = b[0] ^ 0x80
+	}
+}
+
+type packer struct {
+	buf []byte
+}
+
+func NewPacker() *packer {
+	return &packer{
+		buf: make([]byte, 0, 64),
+	}
+}
+
+func NewPackerArray(size int) []*packer {
+	packerArr := make([]*packer, size)
+	for num := range packerArr {
+		packerArr[num] = &packer{
+			buf: make([]byte, 0, 64),
+		}
+	}
+	return packerArr
+}
+
+func (p *packer) putByte(b byte) {
+	p.buf = append(p.buf, b)
+}
+
+func (p *packer) putBytes(b []byte) {
+	p.buf = append(p.buf, b...)
+}
+
+func (p *packer) putBytesNil(b []byte, i int) {
+	for i >= 0 {
+		p.putBytes(b[:i+1])
+		p.putByte(0xFF)
+		b = b[i+1:]
+		i = bytes.IndexByte(b, 0x00)
+	}
+	p.putBytes(b)
+}
+
+func (p *packer) encodeBytes(code byte, b []byte) {
+	p.putByte(code)
+	if i := bytes.IndexByte(b, 0x00); i >= 0 {
+		p.putBytesNil(b, i)
+	} else {
+		p.putBytes(b)
+	}
+	p.putByte(0x00)
+}
+
+func (p *packer) encodeUint(i uint64) {
+	if i == 0 {
+		p.putByte(intZeroCode)
+		return
+	}
+
+	n := bisectLeft(i)
+	var scratch [8]byte
+
+	p.putByte(byte(intZeroCode + n))
+	binary.BigEndian.PutUint64(scratch[:], i)
```

Comment:
Are you sure it is BigEndian?   I think it is LittleEndian ...

Reply:
yes, I'm sure it is bigendian